gateway/ln-gateway/src/ng/tests.rs

```diff
@@ -110,8 +110,19 @@ async fn test_gateway_client_pay_valid_invoice() -> anyhow::Result<()> {
                 .await?
                 .into_stream();
             assert_eq!(gw_pay_sub.ok().await?, GatewayExtPayStates::Created);
-            assert_eq!(gw_pay_sub.ok().await?, GatewayExtPayStates::Preimage);
-            assert_eq!(gw_pay_sub.ok().await?, GatewayExtPayStates::Success);
+            match gw_pay_sub.ok().await? {
+                GatewayExtPayStates::Preimage { preimage: _ } => {}
+                _ => {
+                    panic!("Unexpected state occurred, expected Preimage");
+                }
+            }
+            match gw_pay_sub.ok().await? {
+                GatewayExtPayStates::Success { preimage: _ } => {}
+                _ => {
+                    panic!("Unexpected state occurred, expected Success");
+                }
+            }
```

Comment:
I think you can use something like `assert_matches!(gw_pay_sub.ok().await?, GatewayExtPayStates::Success { .. });`